### PR TITLE
Add action-eslint and lint-action

### PR DESCRIPTION
**Pull request by Charl1996**
This adds the new linter(s) to the repo, lint-action and action-eslint.

See [this commcarehq PR](https://github.com/dimagi/commcare-hq/pull/33280) as example usage. 

In addition to the linked PR I also verified the following scenarios:

1. No .py or .js file changes should not run the linters (the jobs does run until it determines there's not relevant files to lint over).
2. If a linter is not run it should green check that particular linting job.
3. If only .js files changed only eslint should run and flake8 should not be invoked (green check it by default).
4. If only .py files changed only flake8 should run and eslint should not be invoked (green check it by default).
5. If a linter fails the PR can still be merged.

### CODE REVIEW

File: .github/workflows/linter.yml [hunk 1]
```diff
@@ -1,31 +1,71 @@
...
+        uses: tj-actions/changed-files@v24
+        with:
+          files: |
+            **/*.py
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        with:
+          python-version: 3.8
```
Comment (millerdev) [inline .github/workflows/linter.yml:34]: commcare-cloud requires Python 3.10. Would it make more sense to use that version?

File: .github/workflows/linter.yml [hunk 2]
```diff
@@ -1,31 +1,71 @@
...
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        uses: wearerequired/lint-action@v2
+        with:
+          flake8: true
+          flake8_auto_fix: false
+          flake8_args: ${{ steps.changed-files.outputs.all_changed_files }}
+          continue_on_error: false
+
+  # If this is in the same step as python linting the annotations don't show up
+  lint-javascript:
```
Comment (esoergel) [inline .github/workflows/linter.yml:50]: Can we drop this section entirely?  I don't believe commcare-cloud contains any javascript files

**Review by esoergel — COMMENTED** [review]
I don't see the linter running on this PR - is it not functioning properly?

File: .github/workflows/linter.yml (hunk 2 above)
Comment (Charl1996) [reply .github/workflows/linter.yml:50]: You're right, I'll make the update.

**Comment by Charl1996** [comment]
> I don't see the linter running on this PR - is it not fully set up yet?

Ah, I see the workflow was disabled manually. I enabled it and pushed a commit to trigger it.

**Review by millerdev — APPROVED** [review]

File: .github/workflows/linter.yml [hunk 3]
```diff
@@ -45,3 +45,4 @@ jobs:
           flake8_auto_fix: false
           flake8_args: ${{ steps.changed-files.outputs.all_changed_files }}
           continue_on_error: false
+
```
Comment (esoergel) [inline .github/workflows/linter.yml:48]: I don't wanna know how this solves the problem :see_no_evil: 
must've been frustrating to figure out

**Review by esoergel — APPROVED** [review]

**Pull request merged**